### PR TITLE
Adding validation for CR cleanup

### PR DESCRIPTION
**Pull request by atrivedi-px**
**What this PR does / why we need it**:
Adds validation for Backup and Restore CR cleanup

**Which issue(s) this PR fixes** (optional)
Closes #PB-5016

**Special notes for your reviewer**:
Aetos Dashboard:
Schedule Validation : https://aetos.pwx.purestorage.com/resultSet/testSetID/466285
Manual Backup CR Validation : https://aetos.pwx.purestorage.com/resultSet/testSetID/466274

### CODE REVIEW

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -5996,3 +6004,59 @@ func CreateKubevirtBackupRuleForAllVMsInNamespace(ctx context.Context, namespace
...
+			if strings.Contains(eachCR, resourceName) {
+				log.Infof("CR found for [%s] under [%s] namespace", allCRs, currentAdminNamespace)
+				return nil, true, fmt.Errorf("CR cleanup validation failed for - [%s]", resourceName)
+			}
+		}
+
+		return nil, false, nil
+	}
+
+	_, err = task.DoRetryWithTimeout(validateCRCleanupInNamespace, 5*time.Minute, 30*time.Second)
```
Comment (trenukarya-px) [inline tests/backup/backup_helper.go:6058]: Please make it 5 or 10 seconds instead of 30.

File: tests/backup/backup_helper.go [hunk 2]
```diff
@@ -1969,9 +1969,11 @@ func ValidateBackup(ctx context.Context, backupName string, orgID string, schedu
 
 	if len(errStrings) > 0 {
 		return fmt.Errorf("ValidateBackup Errors: {%s}", strings.Join(errStrings, "}\n{"))
-	} else {
-		return nil
 	}
+
+	err = validateCRCleanup(backupName, theBackup.Cluster, theBackup.OrgId, theBackup.Namespaces, ctx, "backup")
```
Comment (trenukarya-px) [inline tests/backup/backup_helper.go:1974]: Do we need to validate anything on application clusters as well in addition to backup application cluster ?

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (atrivedi-px) [reply tests/backup/backup_helper.go]: Changed to 5 seconds.

File: tests/backup/backup_helper.go (hunk 2 above)
Comment (atrivedi-px) [reply tests/backup/backup_helper.go:1974]: We are validating the restore CR cleanup on application/restore clusters

File: tests/backup/backup_helper.go [hunk 3]
```diff
@@ -5996,3 +6004,59 @@ func CreateKubevirtBackupRuleForAllVMsInNamespace(ctx context.Context, namespace
 
 	return ruleStatus, ruleName, nil
 }
+
+func validateCRCleanup(resourceName string, clusterName string, orgID string,
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:6008]: I would suggest having 2 different functions called `ValidateRestoreCRCleanup` and `ValidateBackupCRCleanup` which will take only restore/backup object

If you want a common function you can create a function called `ValidateCRCleanup` which will take an interface. If the interface is backup object the you will call `ValidateBackupCRCleanup` else you will call `ValidateRestoreCRCleanup`

Comment (atrivedi-px) [reply tests/backup/backup_helper.go:6039]: Created ValidateRestoreCRCleanup and ValidateBackupCRCleanup wrapper methods with restore and backup object as input.

**Comment by atrivedi-px** [comment]
Updated Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/468768

**Comment by atrivedi-px** [comment]
Updated Aetos: https://aetos.pwx.purestorage.com/resultSet/testSetID/469802

**Review by trenukarya-px — APPROVED** [review]

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Comment by atrivedi-px** [comment]
Ran full pipeline with this change: https://aetos.pwx.purestorage.com/resultSet/testSetID/470325
Failure is not related to the PR

**Pull request merged**